File: problemas_haskell/problema6.hs
-- Problema 6: Género
grupoGenero :: String -> Int -> String
grupoGenero nombre genero
    | genero == 1 = nombre ++ " es hombre."
    | genero == 2 = nombre ++ " es mujer."
    | genero == 3 = nombre ++ " es No binarie."
    | otherwise   = nombre ++ " tiene otro género."

main :: IO ()
main = do
    putStrLn "Ingrese su nombre:"
    nombre <- getLine
    putStrLn "Ingrese su género (1 para masculino, 2 para femenino, 3 No binarie, 4 otro):"
    generoStr <- getLine
    let genero = read generoStr :: Int
    putStrLn (grupoGenero nombre genero)

File: Problemas_python/problema6.py

nombre = input("Ingrese su nombre: ")
genero = int(input("Ingrese su género (1 para masculino, 2 para femenino, 3 para indeterminado, 4 otro): "))

if genero == 1:
    print(f"{nombre} es hombre.")
elif genero == 2:
    print(f"{nombre} es mujer.")
elif genero == 3:
    print(f"{nombre} eres indeterminado.")
else:
    print(f"{nombre} tiene otro genero.")